ci/do_ci.sh

```diff
@@ -229,6 +229,19 @@ elif [[ "$1" == "cmake.c++20.test" ]]; then
   cmake ${CMAKE_OPTIONS[@]}  \
         -DCMAKE_CXX_FLAGS="-Werror $CXXFLAGS" \
         -DWITH_ASYNC_EXPORT_PREVIEW=ON \
+        -DWITH_STL=CXX20 \
+        ${IWYU} \
+        "${SRC_DIR}"
+  eval "$MAKE_COMMAND"
+  make test
+  exit 0
+elif [[ "$1" == "cmake.c++23.test" ]]; then
+  cd "${BUILD_DIR}"
+  rm -rf *
+  cmake ${CMAKE_OPTIONS[@]}  \
+        -DCMAKE_CXX_FLAGS="-Werror $CXXFLAGS" \
+        -DWITH_ASYNC_EXPORT_PREVIEW=ON \
+        -DWITH_STL=CXX23 \
```

Comment:
[`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) is a target property, not a environment variable in cmake.We can add `-DCMAKE_CXX_STANDARD=23` here to tell cmake to use c++23 by default.

And we need to upgrade cmake to 3.20 or upper to use C++23.The `./ci/setup_cmake.sh` use 3.16 now.

Reply:
The point about `CXX_STANRDARD` seems to be redundant to the other comment.

As for upgrading CMake; how about I just comment out the C++23 section in the YAML file with a new referencing an issue that that as a TODO? I don't need C++23, I just added it out of completeness.

Reply:
Sorry, I forgot we just set `CMAKE_CXX_STANDARD` by `CXX_STANRDARD` in `do_ci.sh` and also upgrade cmake by `CMAKE_VERSION`.Please just ignore my comment.